src/snmalloc/backend/globalconfig.h

```diff
@@ -5,35 +5,65 @@
 #ifndef SNMALLOC_PROVIDE_OWN_CONFIG
 
 #  include "../backend/backend.h"
+#  include "meta_protected_range.h"
+#  include "standard_range.h"
+
+#  if defined(SNMALLOC_CHECK_CLIENT) && !defined(OPEN_ENCLAVE)
+/**
+ * Protect meta data blocks by allocating separate from chunks for
+ * user allocations. This involves leaving gaps in address space.
+ * This is less efficient, so should only be applied for the checked
+ * build.
+ *
+ * On Open Enclave the address space is limited, so we disable this
+ * feature.
+ */
+#    define SNMALLOC_META_PROTECTED
+#  endif
 
 namespace snmalloc
 {
   // Forward reference to thread local cleanup.
   void register_clean_up();
 
-#  ifdef USE_SNMALLOC_STATS
-  inline static void print_stats()
-  {
-    printf("No Stats yet!");
-    // Stats s;
-    // current_alloc_pool()->aggregate_stats(s);
-    // s.print<Alloc>(std::cout);
-  }
-#  endif
-
   /**
    * The default configuration for a global snmalloc.  This allocates memory
    * from the operating system and expects to manage memory anywhere in the
    * address space.
    */
-  class Globals final : public BackendAllocator<Pal, false>
+  class Globals final : public CommonConfig
   {
   public:
+    using Pal = DefaultPal;
     using GlobalPoolState = PoolState<CoreAllocator<Globals>>;
 
   private:
-    using Backend = BackendAllocator<Pal, false>;
+    using ConcretePagemap =
+      FlatPagemap<MIN_CHUNK_BITS, PageMapEntry, Pal, false>;
+
+    using Pagemap = BasicPagemap<Pal, ConcretePagemap, PageMapEntry, false>;
+
+#  if defined(_WIN32) || defined(__CHERI_PURE_CAPABILITY__)
+    static constexpr bool CONSOLIDATE_PAL_ALLOCS = false;
```

Comment:
I think this is moved code, rather than new code, but it might be a good idea to make this a thing that either the PAL or AAL has the option of setting to false and initialising it as something like `pal_feature<ConsolidateAllocs> && aal_feature<ConsolidateAllocs>`

Reply:
Can we punt that to a subsequent PR?